### PR TITLE
[ec2_vpc_nat_gateway] Fix broken check_mode

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY
Fix broken check_mode
-  fix broken check_mode
- comment out integration tests run in check_mode

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ec2_vpc_nat_gateway

### CODE REVIEW

**Review by jillr — COMMENTED** [review]
Just a couple small things, thanks very much @alinabuzachis!

File: plugins/modules/ec2_vpc_nat_gateway.py [hunk 1]
```diff
@@ -990,6 +950,10 @@ def ensure_tags(client, module, nat_gw_id, tags, purge_tags, check_mode):
     final_tags = []
     changed = False
 
+    if check_mode and nat_gw_id is None:
+        # We can't describe tags without an option id, we might get here when creating a new option set in check_mode
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply plugins/modules/ec2_vpc_nat_gateway.py]: @jillr thank you for reviewing. I applied this change.

File: tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml [hunk 2]
```diff
@@ -129,14 +115,25 @@
...
+      that:
+        - create_ngw.changed
+
   - name: Create new NAT gateway with eip allocation-id
     ec2_vpc_nat_gateway:
       subnet_id: "{{ subnet_id }}"
       allocation_id: "{{ allocation_id }}"
       wait: yes
     register: create_ngw
-    retries: 10
```
[the remark this answers is not in the document]
Comment (alinabuzachis) [reply tests/integration/targets/ec2_vpc_nat_gateway/tasks/main.yml:138]: @jillr thank you for reviewing. You are right. I forgot to push the new changes where I removed all the until loops.

**Comment by jillr** [comment]
recheck

**Review by tremble — APPROVED** [review]
LGTM

**Comment by tremble** [comment]
Tests passed.  (initial failures were due to exceeding the AWS quota for EIPs)
Merging.

**Pull request merged**